workflow/executor/resource.go

```diff
@@ -348,37 +349,48 @@ func (we *WorkflowExecutor) SaveResourceParameters(ctx context.Context, resource
 			we.Template.Outputs.Parameters[i].Value = wfv1.AnyStringPtr(output)
 			continue
 		}
-		var cmd *exec.Cmd
+		var args []string
+		var cmdName string
 		if param.ValueFrom.JSONPath != "" {
-			args := []string{"get", resourceName, "-o", fmt.Sprintf("jsonpath=%s", param.ValueFrom.JSONPath)}
+			args = []string{"get", resourceName, "-o", fmt.Sprintf("jsonpath=%s", param.ValueFrom.JSONPath)}
 			if resourceNamespace != "" {
 				args = append(args, "-n", resourceNamespace)
 			}
-			cmd = exec.Command("kubectl", args...)
+			cmdName = "kubectl"
 		} else if param.ValueFrom.JQFilter != "" {
 			resArgs := []string{resourceName}
 			if resourceNamespace != "" {
 				resArgs = append(resArgs, "-n", resourceNamespace)
 			}
 			cmdStr := fmt.Sprintf("kubectl get %s -o json | jq -rc '%s'", strings.Join(resArgs, " "), param.ValueFrom.JQFilter)
-			cmd = exec.Command("sh", "-c", cmdStr)
+			args = append(args, "-c", cmdStr)
+			cmdName = "sh"
 		} else {
 			continue
 		}
-		log.Info(cmd.Args)
-		out, err := cmd.Output()
-		if err != nil {
-			// We have a default value to use instead of returning an error
-			if param.ValueFrom.Default != nil {
-				out = []byte(param.ValueFrom.Default.String())
-			} else {
-				if exErr, ok := err.(*exec.ExitError); ok {
-					log.Errorf("`%s` stderr:\n%s", cmd.Args, string(exErr.Stderr))
+		var output string
+		err := waitutil.Backoff(ExecutorRetry, func() (bool, error) {
```

Comment:
@alexec This pattern is commonly seen throughout the codebase and `waitutil.Backoff`+`argoerr.IsTransientErr` are covered in unit tests.  However there's no e2e test yet unless I miss anything. ` examples/k8s-jobs.yaml` exercises part of this code path but not the transient error part. Is this sufficient? Any suggestions would be appreciated.